File: parciales/ED1-Parcial2-2021-1--002/codigo/Grafos.java
class Grafos {
  public static int colorear(boolean[][] grafo) {
    int n = grafo.length;
    int[] color = new int[n + 1];
    for(int i = 0; i < n; ++i) 
      color[i] = -1;
    color[0] = 0;
    boolean[] colordisponible = new boolean[n + 1];
    for(int i = 0; i < n; ++i) 
      colordisponible[i] = true;
    for(int u = 1; u < n; ++u) {
      for(int i = 0; i < n; ++i)
        if(grafo[u][i])
          if(color[i] != -1)
            colordisponible[color[i]] = false;
      int aval;
      for(aval = 0; aval < n; ++aval)
       if(colordisponible[aval])
         break;
      color[u] = aval; 
    }
    for(int i = 0; i < n; ++i) 
      colordisponible[i] = true;
    boolean[] usado = new boolean[n + 1];
    int res = 0;
    for(int i = 0; i < n; ++i) 
      if(!usado[color[i]]){
        usado[color[i]] = true; 
        res++;
      }
    return res+1;
  }

  public static void main(String[] args){
   boolean[][] g = { {false, true, false}, {false, false, true}, {true, false, false}};
    System.out.println(colorear(g));
  }
}

File: parciales/ED1-Parcial2-2021-1--002/codigo/Grafos.py
import numpy as np
def colorear(grafo):
    n = len(grafo)
    color = np.zeros(n+1)
    for i in range(0,n):
      color[i] = -1
    color[0] = 0
    colordisponible = np.zeros(n + 1)
    for i in range(0,n):
      colordisponible[i] = 1
    for u in range (1,n):
      for i in range(0,n):
        if grafo[u][i] == 1:
          if color[i] != -1:
            colordisponible[int(color[i])] = 0
      color[u] = 0
      for aval in range(0,n):
       if colordisponible[aval] == 1:
         color[u] = aval
         break     
    for i in range(0,n):
      colordisponible[i] = 1
    usado = np.zeros(n+1)
    res = 0
    for i in range(0,n):
      if usado[int(color[i])] == 0:
        usado[int(color[i])] = 1 
        res = res +1
    return res+1

def __main__():
   g = [ [0, 1, 0], [0, 0, 1], [1, 0, 0]]
   print(colorear(g))

__main__()
